### PR TITLE
fix: [CI-10085]: newLogBaseKey prefix support

### PR DESCRIPTION
**Pull request by ShobhitSingh11**
## Describe your changes
Tested with this prefix: kmpySmUISimoRrJL6NL73w%2Fpipeline%2FMacHosted%2F139%2F-8xxwSsnqTcmvv_e6FUu37Q%2FBuild%2FPlugin_1
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54840)

### CODE REVIEW

File: product/log-service/handler/middleware.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ import (
 const authHeader = "X-Harness-Token"
 const authAPIKeyHeader = "x-api-key"
 const routingIDparam = "routingId"
+const regexp1 = "runSequence:[\\d+]"
+const regexp2 = `\w+\/pipeline\/\w+\/[1-9]\d*\/`
```
Comment (smjt-h) [inline product/log-service/handler/middleware.go:31]: Will this work for a larger value of runsequence say 1000?

Comment (ShobhitSingh11) [reply product/log-service/handler/middleware.go:31]: Yes, giving the same result as of above mentioned prefix in description in postman

File: product/log-service/handler/middleware.go [hunk 2]
```diff
@@ -269,7 +271,9 @@ func ValidatePrefixRequest() func(handler http.Handler) http.Handler {
 				return
 			}
 
-			containRunSequence, err := regexp.MatchString("runSequence:[\\d+]", unescapedUrl)
+			containRunSequence, err := regexp.MatchString(regexp1, unescapedUrl)
```
Comment (vistaarjuneja) [inline product/log-service/handler/middleware.go:274]: check for error right after this

**Comment by smjt-h** [comment]
Can we also add a check on number of keys for that prefix should not be more than X.
We can take X=1500
https://github.com/harness/harness-core/blob/830cce4a6ac3546b559ccd51b53bb1bbe9451800/product/log-service/handler/store.go#L237

File: product/log-service/handler/store.go [hunk 3]
```diff
@@ -244,6 +247,16 @@ func HandleZipLinkPrefix(q queue.Queue, s store.Store, c cache.Cache, cfg config
 			return
 		}
 
+		if len(out) > maxItemsToDownload {
+		    err := errors.New("Amount of data is too large to download")
+        	logger.FromRequest(r).
+        	    WithError(err).
+        	    WithField(usePrefixParam, prefix).
+        	    Errorln("api: Cannot download large volume of data")
+        	WriteInternalError(w, fmt.Errorf("Download failed!. Amount of data is too large"))
```
Comment (smjt-h) [inline product/log-service/handler/store.go:256]: error message we can refine as Prefix Key Exceeds Maximum Download Limit

**Review by smjt-h — DISMISSED** [review]

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger messagecheck

**Review by smjt-h — APPROVED** [review]

**Pull request merged**